spec/support/jazz.rb

```diff
@@ -0,0 +1,193 @@
+  end
+
+  # Here's a new-style GraphQL type definition
+  class Ensemble < BaseObject
+    implements GloballyIdentifiable
+    description "A group of musicians playing together"
+    config :config, :configged
+    field :name, "String", null: false
+    field :musicians, "[Jazz::Musician]", null: false
+    # Test extra arguments:
+    field :upcaseName, String, null: false, upcase: true
+
+    def upcase_name
+      @object.name # upcase is applied by the superclass
+    end
+  end
+
+  class Family < GraphQL::Enum
+    description "Groups of musical instruments"
+    # support string and symbol
+    value "STRING", "Makes a sound by vibrating strings", value: :str
+    value :WOODWIND, "Makes a sound by vibrating air in a pipe"
+    value :BRASS, "Makes a sound by amplifying the sound of buzzing lips"
+    value "PERCUSSION", "Makes a sound by hitting something that vibrates"
+    value "KEYS", "Neither here nor there, really"
+    value "DIDGERIDOO", "Makes a sound by amplifying the sound of buzzing lips", deprecation_reason: "Merged into BRASS"
+  end
+
+  # Lives side-by-side with an old-style definition
+  using GraphQL::DeprecatedDSL # for ! and types[]
+  InstrumentType = GraphQL::ObjectType.define do
+    name "Instrument"
+    implements GloballyIdentifiable
+    field :id, !types.ID, "A unique identifier for this object", resolve: ->(obj, args, ctx) { GloballyIdentifiable.to_id(obj) }
+    field :name, !types.String
+    if RUBY_ENGINE == "jruby"
+      # JRuby doesn't support refinements, so the `using` above won't work
+      field :family, Family.to_non_null_type
+    else
+      field :family, !Family
+    end
+  end
+
+  class Musician < BaseObject
+    implements GloballyIdentifiable
+    description "Someone who plays an instrument"
+    field :name, String, null: false
+    field :instrument, InstrumentType, null: false
+  end
+
+  # Another new-style definition, with method overrides
+  class Query < BaseObject
+    field :ensembles, [Ensemble], null: false
+    field :find, GloballyIdentifiable, null: true do
+      argument :id, "ID", null: false
+    end
+    field :instruments, [InstrumentType], null: false do
+      argument :family, Family, null: true
+    end
+
+    def ensembles
+      Models.data["Ensemble"]
+    end
+
+    def find(id:)
+      GloballyIdentifiable.find(id)
+    end
+
+    def instruments(family: nil)
+      objs = Models.data["Instrument"]
+      if family
+        objs = objs.select { |i| i.family == family }
+      end
+      objs
+    end
+  end
+
+  class EnsembleInput < GraphQL::InputObject
+    argument :name, String, null: false
+  end
+
+  class Mutation < BaseObject
+    field :addEnsemble, Ensemble, null: false do
+      argument :input, EnsembleInput, null: false
+    end
+
+    def add_ensemble(input:)
+      # TODO, how should this object be presented here?
+      # Maybe an instance of the class above, whose methods may be called?
+      ens = Models::Ensemble.new(input["name"])
```

Comment:
Should we prefer the use of symbols here? i.e. `input[:name]` instead.

Reply:
😆  yes i suppose you're right. 

Maybe I should drop Ruby 2.1 support so I can `.to_sym` willy-nilly